File: Minimum Integer.py
from typing import List


class Solution:
    def minimumInteger(self, N : int, A : List[int]) -> int:
        # code here
        sumo = sum(A)
        ans = 10**9
        for x in A:
            if N*x >= sumo:
                ans = min(ans, x)
        return ans
        

File: Minimum Integer.cpp
class Solution {
  public:
    int minimumInteger(int N, vector<int> &A) {
        // code here
        int mmin = 1e9;
        long long sum = 0;
        for(int i:A)sum += i;
        for (int i:A)
        {
            if (i*1LL*N >= sum)
            {
                mmin = min(mmin, i);
            }
        }
        return mmin;
    }
};
